### ISSUE TITLE
Improved character creation

### ISSUE THREAD

**Issue by anhegofcherek** [prophecymud/ideas#108]
It's not been touched in quite some time, and there's definitely room for improvement.

I don't necessarily love the idea of creating an unscoped megathread, but there's a lot of stuff that could be done here, from big to small, and here seems like a decent place to aggregate and discuss it.

So, chime in with suggestions please!

**Comment by anhegofcherek** [comment]
The thing that kicked this all off: removing hometown "choice" (since there's only one) from creation

There's only been one for years, and the current plan to add more (#48) has _starting_ hometowns scoped for much later, so even with an improved hometown system, people would still start at Sendar

**Comment by CorlanProphecy** [comment]
There should be a pre-set suggested build for each class. Making a decent character is pretty much impossible for a new player!

**Comment by grekan** [comment]
Some idea that come to mind:

- Removing hometown option would be great. 

- Picking your god seems silly, it has little to no impact on the game that I can tell and stops you from seeing races/classes right away. If keeping it is needed I would say allow any race/class to pick any god, there will be turn coats and weirdos in every place. 

- Spiffing the defaults would be nice. Perhaps 2k/level is the base instead of 1k? Or 1.5. It would be great if default gave no more or less points than adding all the individual skills (it may already do this). This would mean new players who skipped customization would not be so handicapped. It could also be more useful during rerolls to just add the default.

- I think it would also be great if a player skipped customization, that they would get transportation thrown on. We have word for all at Level 1, but it still requires transportation spells (tested this today). Alternatively, break word out as a separate skill/spell in creation and it comes in transportation or in the default skillset. 

- Any gender combo for race/class - with possible exceptions for RP combos such as knights, dryads, horsemen, berserkers. That being said, it's 2020 and Eddings' work is a bit dated. Taking some liberties with the RP is not necessarily a bad thing. 

- Have align be an option you can pick instead of it being based on god/race/class. 

- Please do keep customization, it would suck to not have. 

- Some kind of way to acknowledge if players are returning players or not to keep track of who is actually new (I think Eirela suggested this one)

- Some kind of name generator?

- Name checks for names similar to the one input? Also a dictionary check? Not sure how doable these are / intensive.

**Comment by Vordai-of-Prophecy** [comment]
I agree with much of what Grekan and Corlan suggested. However, the chance of a name generator coming up with a Prophecy-appropriate name every time is pretty slim. It will definitely aggravate new players to select a name this way and then have it rejected for validation -- whether it's because there's already a similar Prophecy player/mob/book character name, or because it pulls 4M Google hits, or whatever. 
I'm assuming that simply relaxing the valid name requirements is not an option...

If there's a way to allow rerolling players an option to keep their original characterization choices (or to at least display them), I think it would be a nice addition. Not everyone rerolls to change their skills/spells -- some of us just trained mana too much, for example.

Also, if you want to keep the God selection, then I suggest listing the applicable races after each God's name (ex. 
Ul       |  Good  |  Ulgo, Dryad  |  The Father of the Gods. Wise and patient.) However, I do also like the idea of "weirdo" God/race/class/alignment combinations!

**Comment by anhegofcherek** [comment]
> Removing hometown option would be great.

Done

**Comment by anhegofcherek** [comment]
> Any gender combo for race/class - with possible exceptions for RP combos such as knights, dryads, horsemen, berserkers.

I don't understand - that's what we have at the moment

> That being said, it's 2020 and Eddings' work is a bit dated

Agreed - I'd happily drop the gender constraints on that basis

**Comment by grekan** [comment]
> > Any gender combo for race/class - with possible exceptions for RP combos such as knights, dryads, horsemen, berserkers.
> 
> I don't understand - that's what we have at the moment

I wasn't thinking straight, disregard that part. Any gender for any race/class would be cool though.

**Comment by anhegofcherek** [comment]
> Please do keep customization, it would suck to not have.

@grekan can we talk more about this in #116 please? Thanks!
